content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1507 @@
+
+To run the reference application found in `content/_code-samples/build-a-wallet/desktop-js` for this step, run:
+
+```console
+npm run account
+```
+
+### 4. Show Account's Transactions
+
+**Full code for this step:**
+[`library/3_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/3_helper.js),
+[`library/4_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/4_helper.js),
+[`4-tx-history/index.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/4_tx-history.js),
+[`4-tx-history/view/preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/4_tx-preload.js),
+[`4-tx-history/view/renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/4_tx-renderer.js),
+[`4-tx-history/view/index.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/4_tx-history.html).
+
+At this point, our wallet shows the account's balance getting updated, but doesn't give us any clue about how this state came about, namely the actual transactions that caused the updates. So, our next step is to display the account's up to date transaction history using subscriptions once again:
+
+![Screenshot: Step 4, show transaction history](img/javascript-wallet-4.png)
+
+1. In the `library` folder, add a new file `4_helpers.js`. Then add the following helper function to that file:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/library/4_helpers.js", language="js") }}
+
+2. Now, in `index.js`, require the new helper function at the bottom of the import section like so:
+
+```javascript
+const { prepareAccountData, prepareLedgerData} = require('./library/3_helpers')
+const { prepareTxData } = require('./library/4_helpers')
+```
+
+3. In `index.js`, update the listener function subscribed to the `transaction` event by adding the following snippet:
+
+```javascript
+// Wait for transaction on subscribed account and re-request account data
+client.on("transaction", async (transaction) => {
+  // Reference: https://xrpl.org/account_info.html
+  const accountInfoRequest = {
+    "command": "account_info",
+    "account": address,
+    "ledger_index": transaction.ledger_index
+  }
+
+  const accountInfoResponse = await client.request(accountInfoRequest)
+  const accountData = prepareAccountData(accountInfoResponse.result.account_data)
+  appWindow.webContents.send('update-account-data', accountData)
+
+  // Step 4 code additions - start
+  const transactions = prepareTxData([{tx: transaction.transaction}])
+  appWindow.webContents.send('update-transaction-data', transactions)
+  // Step 4 code additions - end
```

Comment:
This part doesn't seem to be enough on its own. Comparing mine to the version of `index.js` in `4-tx-history/`, I'm missing the "Initial Transaction Request" section (the steps as called out in the tutorial never have you add it) and I'm not sure if that or something else is why I was getting a `ReferenceError: transaction is not defined` error at this stage, and the transaction table not populating.

Reply:
It vanished from the tutorial somehow, I've re-added it